contexts/pathogen-context/pathogen-context.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ToastProvider } from "../toast-provider";
+import { MapProvider, MapRef, useMap } from "react-map-gl";
+import React, { Context, Dispatch, createContext, useEffect, useReducer } from "react";
+import { filterData } from "./filter-update-steps/apply-new-selected-filters";
+import { handleFilterUpdate } from "./filter-update-steps";
+import { useArboData } from "@/hooks/useArboData";
+
+export interface PathogenContextType<TData extends Record<string, unknown>> extends PathogenContextState<TData> {
+  dispatch: React.Dispatch<PathogenContextAction>;
+}
+
+export interface PathogenContextState<TData extends Record<string, unknown>> {
+  filteredData: TData[];
+  selectedFilters: { [key: string]: string[] };
+  dataFiltered: boolean;
+}
+
+export interface PathogenContextAction {
+  type: PathogenContextActionType;
+  payload: any;
```

Comment:
Generic is a big step up and means that SC2 tracker won't have `any`s everywhere in the same way arbotracker does but I didn't perfect this type yet. In my perfect world this would be a union type with more strict typing depending on which action type it is, but again one thing at a time.